Nano 33 BLE - enable the SPI1 sensors

KurtE opened the pull request:
Resolves: #51

There is an IO pin on the NANO that that when turned on, enables most of the sensors on the BLE sense which are on SPI1.   The MBED version, enables this pin as part of the main().

Which I am trying to emulate.  There is code already in, that if you use at least one of the zephyr device drivers, will eanble this pin.  However that does not help when you are using external libraries or the like.

So I added details about this pin in our overlay file, in the zephyr,user section.

I then added code to main.cpp, that is only included if your are building using an NRFX board.  Currently the nano and one of the nicla boards.  Could also specically only do this on the NANO, but probably does not matter as, the code tries to find that property and only if it is found, does it turn on the pin.

Note: The MBED version turn on this pin with high drive.  Which I emulated using the information, mentioned in the zephyr discussion.
https://github.com/zephyrproject-rtos/zephyr/discussions/78710 In one of my sketches I verified that the pin pads configuration looks the same as mbed setup.

With these changes I am able to access most of the sensors. Most of the testing has been done by @mjs513, with some by myself as well.

facchinm reviewed (changes requested):
I would prefer to move this in `initVariant()` inside the Nano 33 BLE variant.cpp, WDYT?

facchinm commented:
> I would prefer to move this in `initVariant()` inside the Nano 33 BLE variant.cpp, WDYT?

To allow this I'm going to add a weak initVariant() in main.cpp as it already happens with all the other cores

KurtE commented:
Sounds good will try it out hopefully later today
Makes sense, to move out all of the core specific things into the variant.

KurtE commented:
I rebased this one and added the call to initVariant to main.cpp, added a week version in there as well that does nothing
And then moved the code into the Nano 33 sense variant.cpp file
